### PR TITLE
DX: integrate PRLint

### PR DESCRIPTION
**Pull request by keradus**
I'm open for changes here, but I wouldn't overthink it

### CODE REVIEW

File: .github/prlint.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "title": [
+    {
+      "pattern": "^(feature|bug|minor|Docs|DX):\\s.+\\S$",
```
Comment (keradus) [inline .github/prlint.json:4]: I'm open for changes here, but I wouldn't overthink it

**Comment by SpacePossum** [comment]
nice idea :)  can we have `Docs` in lowercase as well so it matches the others ?(besides `DX` which is great in all upper)

File: .github/prlint.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+      "message": "Your title needs to match the convention: use '(feature|bug|minor|Docs|DX):' prefix."
```
Comment (keradus) [inline .github/prlint.json:5]:
```suggestion
      "pattern": "^(feature|bug|minor|docs|DX):\\s.+\\S$",
      "message": "Your title needs to match the convention: use '(feature|bug|minor|docs|DX):' prefix."
```

**Review by SpacePossum — APPROVED** [review]

**Review by kubawerlos — APPROVED** [review]

**Review by julienfalque — APPROVED** [review]

**Comment by keradus** [comment]
@SpacePossum @localheinz @kubawerlos @julienfalque , this was the last step to stop using internal `gh` cli tool , now we can rely on github merge button 🎉

**Comment by SpacePossum** [comment]
nice one :)    thanks so much!
curious about the change log generation for the releases without `gh`, but not something I worry about though, just curious :+1:

**Comment by localheinz** [comment]
@SpacePossum 

Release automation is fairly easy, see:

- https://github.com/ergebnis/.github/blob/ad9e0939fc1ca80b8de4e7f4f102e834f3d17f63/.github/workflows/release.yaml
- https://github.com/ergebnis/.github/blob/main/actions/github/release/create/action.yaml

**Comment by SpacePossum** [comment]
might be worth updating the regex `DX - cleanup test ` is currently not allowed

**Comment by keradus** [comment]
actually, the changelog was always `git log` (with fancy filters etc) and not `gh`.

for release, we also bump internal code marker (eg version, codename) and trigger new release for phar repo, as well as trigger website regeneration. I would say it's another scope that has one blocker - I'm not comfortable to share my secret to sign phar file to CI, after issue in Travis that allowed to expose the secrets stored in CI system.

**Pull request merged**